src/com/elikill58/negativity/spigot/packets/PacketManager.java

```diff
@@ -1,21 +1,52 @@
 package com.elikill58.negativity.spigot.packets;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.bukkit.Bukkit;
-import org.bukkit.plugin.Plugin;
+import org.bukkit.entity.Player;
 
-import com.elikill58.negativity.spigot.packets.protocollib.ProtocollibSupport;
+import com.elikill58.negativity.spigot.SpigotNegativity;
+import com.elikill58.negativity.spigot.packets.event.PacketEvent.PacketSourceType;
+import com.elikill58.negativity.spigot.packets.event.PacketReceiveEvent;
+import com.elikill58.negativity.spigot.packets.event.PacketSendEvent;
 
-public class PacketManager {
-	
-	public static boolean hasProtocollib = false;
+public abstract class PacketManager {
+
+	public abstract void addPlayer(Player p);
+	public abstract void removePlayer(Player p);
+	public abstract void clear();
+
+	final List<PacketHandler> handlers = new ArrayList<>();
```

Comment:
```suggestion
	private final List<PacketHandler> handlers = new ArrayList<>();
```

Dernier truc et je ne t'embêtes plus avec cette classe.